tests/test_create.py

```diff
@@ -186,3 +186,12 @@ def test_get_repo_with_token_returns_same_url_with_no_token():
 def test_appended_github_url_not_replaced():
     repo = "https://github.com.someothersite.net/opensafely-test/my-test-repo"
     assert create.get_repo_with_token(repo, "a_secure_tken") == repo
+
+
+def test_repo_with_username_password_replaced():
+    repo = "https://username:password@github.com/opensafely-test/my-test-repo"
+    expected_repo = (
+        "https://interactive:a_secure_token@github.com/opensafely-test/my-test-repo"
+    )
+
+    assert create.get_repo_with_token(repo, "a_secure_token") == expected_repo
```

Comment:
This is the same test as `test_appended_github_url_not_replaced`, just with different parameters. Having reconsidered the name of `test_appended_github_url_not_replaced`, could you then consider [parameterizing](https://docs.pytest.org/en/latest/how-to/parametrize.html) it?

Reply:
That said, I don't understand the purpose of this test.